rustls-mbedcrypto-provider/src/kx.rs

```diff
@@ -209,22 +249,41 @@ impl ActiveKeyExchange for KeyExchange {
     }
 }
 
-#[derive(Debug)]
-pub(crate) struct DheKxGroup {
-    pub(crate) named_group: NamedGroup,
-    pub(crate) group: FfdheGroup<'static>,
-    pub(crate) priv_key_len: usize,
+/// A DHE key-exchange group supported by *mbedtls*.
+///
+/// All possible instances of this type are provided by the library in
+/// the `ALL_KX_GROUPS` array.
+pub struct DheKxGroup<T: RngCallback> {
```

Comment:
Can we not make all the fields public, and instead add a `with_rng_provider` function? This would also make it easier to use (and less error-prone). This also applies to `KxGroup`

Reply:
Let me try

Reply:
@s-arash , Just add the function and also renamed rng_provider . Please take another look. thx!